source/slang/slang-diagnostic-defs.h

```diff
@@ -524,6 +524,14 @@ DIAGNOSTIC(
     Error,
     missingLayoutBindingModifier,
     "Expecting 'binding' modifier in the layout qualifier here")
+DIAGNOSTIC(
+    20017,
+    Error,
+    constNotAllowedOnCStylePtrDecl,
+    "'const' not allowed on pointer typed declarations using the C style '*' operator. "
+    "If the intent is to restrict the pointed-to value to read-only, use 'Ptr<T, Access.Read>' "
+    "instead; if the intent is to make the pointer itself immutable, use 'const Ptr<...>'.")
```

Comment:
We really shouldn't be encouraging people to use `const` at all, even in the latter case (where they want an immutable pointer). It would be great if the second part of this message could say something like "if the intent is to have an immutable variable of pointer type, use `let` to declare the variable."